src/pools/abstract-pool.ts

```diff
@@ -0,0 +1,173 @@
+import EventEmitter from 'events'
+import type { MessageValue } from '../utility-types'
+import type { IPool } from './pool'
+
+export type ErrorHandler<Worker> = (this: Worker, e: Error) => void
+export type OnlineHandler<Worker> = (this: Worker) => void
+export type ExitHandler<Worker> = (this: Worker, code: number) => void
+
+export interface IWorker {
+  on(event: 'error', handler: ErrorHandler<this>): void
+  on(event: 'online', handler: OnlineHandler<this>): void
+  on(event: 'exit', handler: ExitHandler<this>): void
+}
+
+export interface PoolOptions<Worker> {
+  /**
+   * A function that will listen for error event on each worker.
+   */
+  errorHandler?: ErrorHandler<Worker>
+  /**
+   * A function that will listen for online event on each worker.
+   */
+  onlineHandler?: OnlineHandler<Worker>
+  /**
+   * A function that will listen for exit event on each worker.
+   */
+  exitHandler?: ExitHandler<Worker>
+  /**
+   * This is just to avoid not useful warnings message, is used to set `maxListeners` on event emitters (workers are event emitters).
+   *
+   * @default 1000
+   */
+  maxTasks?: number
+}
+
+class PoolEmitter extends EventEmitter {}
+
+export abstract class AbstractPool<
+  Worker extends IWorker,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Data = any,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Response = any
+> implements IPool<Data, Response> {
+  public readonly workers: Worker[] = []
+  public nextWorker: number = 0
+
+  /**
+   * `workerId` as key and an integer value
```

Comment:
`an integer value` doesn't help with understanding what this is used for

Reply:
Good catch, this was already present it is my bad.
The key is the worker and the value is the number of tasks that has been assigned to that worker since it started.

This is really used only to set to be used on the setMaxEventListeners, I don't remember if is used for other scopes....maybe not so we could potentially removed.

I guess when I added this map I was preparing something for the future like max number of active tasks into a thread( We need to think if we want to add this later on or not).
If you can please create an issue with those points would be great